src/App.tsx

```diff
@@ -1,26 +1,151 @@
+      .getTodos()
+      .then(setTodoList)
+      .catch(() => {
+        setErrorMessage(ErrorMessages.getError);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const deleteTodos = (todoId: number) => {
+    setLoadingIds(ids => [...ids, todoId]);
+    todoSetvices
+      .deletePost(todoId)
+      .then(() => {
+        setTodoList(currentTodo =>
+          currentTodo.filter(todo => todo.id !== todoId),
+        );
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.deleteError);
+      })
+      .finally(() => {
+        setLoadingIds(ids => ids.filter(id => id !== todoId));
+        inputRef.current?.focus();
+      });
+  };
+
+  const addPost = (title: string) => {
+    const newTitle = title.trim();
+
+    if (newTitle.trim() === '') {
+      return setErrorMessage(ErrorMessages.emptyTitleError);
+    }
+
+    const tempNewTodo: Todo = {
+      id: 0,
+      userId: todoSetvices.USER_ID,
+      title: newTitle,
+      completed: false,
+    };
+
+    setTempTodo(tempNewTodo);
+    setIsLoading(true);
+    setLoadingIds(currentIds => [...currentIds, 0]);
+
+    return todoSetvices
+      .addPost(newTitle)
+      .then(newTodo => {
+        setTodoList(currentTodos => [...currentTodos, newTodo]);
+        setInputValue('');
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorMessages.addError);
+        setInputValue(newTitle);
+        setTempTodo(null);
+      })
+      .finally(() => {
+        setIsLoading(false);
+        setLoadingIds([]);
+      });
+  };
+
+  useEffect(() => {
+    getTodoFromServer();
+  }, []);
+
+  const areAllCompleted =
+    todoList.length > 0 && todoList.every(todo => todo.completed);
+  // if (!USER_ID) {
+  //   return <UserWarning />;
+  // }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          areAllCompleted={areAllCompleted}
+          addPost={addPost}
+          isLoading={isLoading}
+          inputValue={inputValue}
+          setInputValue={setInputValue}
+          inputRef={inputRef}
+        />
+
+        <TodoList
+          isLoading={isLoading}
+          todoList={todoList}
+          currentFilter={currentFilter}
+          deleteTodos={deleteTodos}
+          deletedIds={loadingIds}
+          tempTodo={tempTodo}
+        />
+
+        {todoList.length > 0 && (
+          <Footer
+            todoList={todoList}
+            currentFilter={currentFilter}
+            onChange={setCurrentFilter}
```

Comment:
```suggestion
            onFilterChange={setCurrentFilter}
```
because onChange looks a bit strange in Footer component